scripts/Modules/Saritasa.Cake/Tasks/PackTask.cs

```diff
@@ -0,0 +1,148 @@
+        {
+            "Saritasa.Tools.Common",
+            "Saritasa.Tools.Domain",
+            "Saritasa.Tools.EntityFrameworkCore",
+            "Saritasa.Tools.Emails",
+            "Saritasa.Tools.Misc",
+        };
+    }
+
+    private string[] projectNamesForPack;
+
+    /// <inheritdoc />
+    public override void Run(PackContext context)
+    {
+        UpdateAssemlyVersions(context);
+
+        PackProjects(context);
+
+        RevertAssemlyVersions();
+    }
+
+    private void UpdateAssemlyVersions(PackContext context)
+    {
+        var revcount = context.ExecuteGitCommand("rev-list --all --count").Replace(Environment.NewLine, string.Empty);
+        var hash = context.ExecuteGitCommand("log --pretty=format:%h -n 1").Replace(Environment.NewLine, string.Empty);
+
+        foreach (var projectName in projectNamesForPack)
+        {
+            var assemblyVersion = GetProjectAssemblyVersion(projectName);
+            var fileVersion = $"{GetVersion(projectName)}.{revcount}";
+            var productVersion = $"{fileVersion}-{hash}";
+            context.Information($"{projectName} has versions {assemblyVersion} {fileVersion} {productVersion}");
+
+            ReplaceAttributeValueInAssemblyInfo(projectName, "AssemblyVersion", assemblyVersion);
+            ReplaceAttributeValueInAssemblyInfo(projectName, "AssemblyFileVersion", fileVersion);
+            ReplaceAttributeValueInAssemblyInfo(projectName, "AssemblyInformationalVersion", productVersion);
+        }
+    }
+
+    private void PackProjects(PackContext context)
+    {
+        foreach (var projectName in projectNamesForPack)
+        {
+            var projectDir = $"{SolutionDir}src\\{projectName}";
+
+            context.DotNetRestore(projectDir);
+            context.DotNetBuild(projectDir, new DotNetBuildSettings
+            {
+                Configuration = context.MsBuildConfiguration
+            });
+
+            PackNuGetFile(context, projectName, projectDir);
+        }
+    }
+
+    private void PackNuGetFile(PackContext context, string projectName, string projectDir)
+    {
+        var longHash = context.ExecuteGitCommand("rev-parse HEAD").Replace(Environment.NewLine, string.Empty);
+        var nuspecFile = $"{projectDir}\\{projectName}.nuspec";
+
+        context.CopyFile($"{nuspecFile}.template", nuspecFile);
+
+        UpdateVariableInFile(nuspecFile, "CommitHash", longHash);
+
+        var assemblyVersion = GetVersion(projectName);
+        var resultPackNuGet = context.ExecuteNuGetCommand($"pack {nuspecFile} " +
+            $"-NonInteractive " +
+            $"-Version {assemblyVersion} " +
+            $"-Exclude .snk " +
+            $"-OutputDirectory {SolutionDir}");
+
+        context.Information(resultPackNuGet);
+    }
+
+    private void RevertAssemlyVersions()
+    {
+        foreach (var projectName in projectNamesForPack)
+        {
+            ReplaceAttributeValueInAssemblyInfo(projectName, "AssemblyVersion", "1.0.0.0");
```

Comment:
Can we have constants for `AssemblyVersion`, `AssemblyFileVersion`, `AssemblyInformationalVersion` to reuse in this file?
Also, not sure if it really "reverts" those values, maybe we should have remembered the original values first and then revert to them? @krasninja

Reply:
Also, we can try this: `dotnet build /p:AssemblyVersion=1.2.3.4`. Not sure if it would work with nuget command.

Reply:
Doesn't look like it's resolved?

Reply:
All our projects contains custom `AssemblyInfo.cs` files (project property `GenerateAssemblyInfo = false`)
`dotnet build /p:AssemblyVersion=1.2.3.4` not apply version, when `GenerateAssemblyInfo = false`
Open issue in dotnet - [issue](https://github.com/dotnet/project-system/issues/3934)
Question in stackoverflow - [answer](https://stackoverflow.com/a/61277331)

Resolved only constants)